parcels/application_kernels/advection.py

```diff
@@ -115,9 +115,15 @@ def AdvectionRK45(particle, fieldset, time):  # pragma: no cover
     Time-step dt is halved if error is larger than fieldset.RK45_tol,
     and doubled if error is smaller than 1/10th of tolerance.
     """
-    dt = min(
-        particle.next_dt / np.timedelta64(1, "s"), fieldset.RK45_max_dt
-    )  # TODO: improve API for converting dt to seconds
+    dt = particle.next_dt / np.timedelta64(1, "s")  # TODO: improve API for converting dt to seconds
+    if dt > fieldset.RK45_max_dt:
+        dt = fieldset.RK45_max_dt
+        particle.next_dt = fieldset.RK45_max_dt * np.timedelta64(1, "s")
+    if dt < fieldset.RK45_min_dt:
+        particle.next_dt = fieldset.RK45_min_dt * np.timedelta64(1, "s")
+        return StatusCode.Repeat
+    particle.dt = particle.next_dt
+
```

Comment:
Is this a change in behaviour from v3, or is it something that was in v3 and is being re-implemented in v4? Perhaps worth noting in the body of the PR description?

Reply:
Yes, this was a (small) bug in RK45 in v3 that I fixed now